micro_sam/sam_annotator/image_series_annotator.py

```diff
@@ -108,21 +121,36 @@ def image_series_annotator(
     state = AnnotatorState()
     state.decoder = decoder
     state.initialize_predictor(
-        image, model_type=model_type, save_path=image_embedding_path,
-        halo=halo, tile_shape=tile_shape, predictor=predictor, ndim=2,
-        precompute_amg_state=precompute_amg_state,
+        image, model_type=model_type, save_path=image_embedding_path, halo=halo, tile_shape=tile_shape,
+        predictor=predictor, ndim=3 if is_volumetric else 2, precompute_amg_state=precompute_amg_state,
         checkpoint_path=checkpoint_path, device=device,
     )
-    state.image_shape = image.shape[:-1] if image.ndim == 3 else image.shape
+    if image.ndim == 2:
+        state.image_shape = image.shape
+    elif image.ndim == 3:
+        if is_volumetric:
+            state.image_shape = image.shape
+        else:
+            state.image_shape = image.shape[:-1]
+    elif image.ndim == 4:
+        state.image_shape = image.shape[:-1]
+
+    if image.ndim == 2:
```

Comment:
You need to use `is_volumetric` here as well. Otherwise we again have the issue that RGB data wrongly triggers the 3d annotator.